recipes/cppcommon/all/patches/00003-define-win32-winnt.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index ad3d53aa8..43418041f 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -74,6 +74,10 @@ target_link_libraries(cppcommon PUBLIC ${LINKLIBS} CONAN_PKG::fmt)
+ list(APPEND INSTALL_TARGETS cppcommon)
+ list(APPEND LINKLIBS cppcommon)
+ 
++if(WIN32)
++  target_compile_definitions (cppcommon PUBLIC -D_WIN32_WINNT=_WIN32_WINNT_WIN8)
```

Comment:
Is windows 8 the lowest windows version having `SYMBOLIC_LINK_FLAG_DIRECTORY`?

Reply:
It's weird. Actually I've got it working with `_WIN32_WINNT_WIN7` with this profile:

```
[settings]
os=Windows
os_build=Windows
compiler=gcc
compiler.version=7
compiler.libcxx=libstdc++11
compiler.threads=posix
arch=x86_64
compiler.exception=seh
[options]
[build_requires]
mingw_installer/1.0@conan/stable
msys2/20190524
[env]
```

But then, my default Visual Studio profile stoped working:

```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.cppstd=17
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
```

The error was:

```
C:\XXX\.conan\data\cppcommon\cci.20201104\_\_\build\748fa6af231d04ec8c6b58b5e0027c19c5c352a1\source_subfolde
r\source\time\timestamp.cpp(114,5): error C3861: 'GetSystemTimePreciseAsFileTime': identifier not found [C:\XXX\.conan\data\cppcommon\cci.20201104\_\_\build\748fa6af231d04ec8c6b58b5e0027c19c5c352a1\build_subfolder\source_subfol
der\cppcommon.vcxproj]
C:\XXX\.conan\data\cppcommon\cci.20201104\_\_\build\748fa6af231d04ec8c6b58b5e0027c19c5c352a1\source_subfolde
r\source\time\timestamp.cpp(175,9): error C3861: 'GetSystemTimePreciseAsFileTime': identifier not found [C:\XXX\.conan\data\cppcommon\cci.20201104\_\_\build\748fa6af231d04ec8c6b58b5e0027c19c5c352a1\build_subfolder\source_subfol
der\cppcommon.vcxproj]
```

So I just increased the define to the minimum prereq of `GetSystemTimePreciseAsFileTime()` (https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getsystemtimepreciseasfiletime).

I didn't investigate why it worked with the mingw profile with `_WIN32_WINNT_WIN7`, but I guess there has to be some conditional ifdef somewhere in the project.

Reply:
did you try including `sdkver.h`? I prefer using the version matched to the SDK rather than just forcing it down

Reply:
Hy, @prince-chrismc . I didn't know of this header!

But it is rather `SDKDDKVer.h` actually, isn't it?

Reply:
yes that definitely it

🤣 I should have googled the correct name but I never remember that I don't remember it

Reply:
I got it working with:

https://github.com/conan-io/conan-center-index/blob/2575fba18faa81bff436782db3fc23392d051536/recipes/cppcommon/all/patches/00003-define-win32-winnt.patch#L9-L15

To include `SDKDDKVer.h` worked only for msvc builds...

Reply:
Actually, as `SDKDDKVer.h` didn't work for mingw builds, I think the right way to do this is to leave MSVC builds alone and set `_WIN32_WINNT=_WIN32_WINNT_VISTA` for everything else. I'm building here locally and will pull it rightway.

Reply:
So, it ended up like this:

https://github.com/conan-io/conan-center-index/blob/111a4c3efb76e61d2b94cc4e6c9c46acd9a4f688/recipes/cppcommon/all/patches/00003-define-win32-winnt.patch#L9-L11

@prince-chrismc, unfortunately `SDKDDKVer.h` was of no avail to the mingw builds =(

@madebr, I was doing something wrong, it seems that `_WIN32_WINNT_VISTA` suffices.